libcodechecker/server/session_manager.py

```diff
@@ -79,30 +67,33 @@ def is_alive(self):
             self.__session_lifetime
 
     def revalidate(self):
-        if self.is_alive:
-            # A session is only revalidated if it has yet to exceed its
-            # lifetime. After a session hasn't been used for this interval,
-            # it can NOT be resurrected at all --- the user needs to log in
-            # to a brand-new session.
-            self.last_access = datetime.now()
-
-            if self.__database:
-                # Update the timestamp in the database for the session's last
-                # access.
-                try:
-                    transaction = self.__database()
-                    record = transaction.query(SessionRecord). \
-                        get(self.persistent_hash)
+        if not self.is_alive:
+            return
 
-                    if record:
-                        record.last_access = self.last_access
-                        transaction.commit()
-                except Exception as e:
-                    LOG.error("Couldn't update usage timestamp of {0}"
-                              .format(self.token))
-                    LOG.error(str(e))
-                finally:
-                    transaction.close()
+        # A session is only revalidated if it has yet to exceed its
```

Comment:
Please move this up as a function comment, to see there in which cases are the revalidation done.